### PR TITLE
Targeted refresh creates resources from other regions based on events

### PR DESCRIPTION
**Pull request by agrare**
Azure is not as region-focused as AWS is, for example you ask for a list of all VMs we have to filter the resulting list by the region we are looking for.

Similarly events are raised across all regions, so if you have an azure footprint that spans regions and you have a MIQ provider per-region, all events will be processed by every provider.

This becomes an issue when using events to trigger targeted refresh, because unlike the full refresh case where we get the full list and filter by region, targeted refresh simply runs a GET on the resource ID representing the target.

The end result is that if you create a VM in one region, an event is raised in every MIQ provider, and targeted refresh is run on that new VM in every provider, meaning you will have a copy of the VM in every MIQ provider.  A full refresh will reset this.

This is only the targeted refresh part of the fix, filtering events by region would be another beneficial change since that would reduce the overall workload of the system however the regions isn't part of the event payload and so it will most likely have to be looked up from the resource group.

Long term we want to get away from one-provider-per-region for exactly this reason.

### CODE REVIEW

File: app/models/manageiq/providers/azure/refresh_helper_methods.rb [hunk 1]
```diff
@@ -84,6 +82,14 @@ def gather_data_for_this_region(arm_service, method_name = 'list_all')
     end
   end
 
+  def in_this_region(resources = nil)
```
Comment (Fryguy) [inline app/models/manageiq/providers/azure/refresh_helper_methods.rb:85]: Something about this reads funny to me - perhaps something like `filter_my_region` or `filtered_by_region`? I'm not sure

File: app/models/manageiq/providers/azure/inventory/collector/target_collection.rb [hunk 2]
```diff
@@ -31,8 +31,10 @@ def resource_groups
                            end
                          else
                            collect_inventory_targeted(:resource_groups) do
-                             Parallel.map(refs, :in_threads => thread_limit) do |ref|
-                               safe_targeted_request { @rgs.get(ref) }
+                             in_this_region do
+                               Parallel.map(refs, :in_threads => thread_limit) do |ref|
```
[the remark this answers is not in the document]
Comment (Fryguy) [reply app/models/manageiq/providers/azure/inventory/collector/target_collection.rb:35]: Also, should the refs be filtered in the region _before_ they are requested?

[the remark this answers is not in the document]
Comment (agrare) [reply app/models/manageiq/providers/azure/inventory/collector/target_collection.rb]: Done

**Comment by Fryguy** [comment]
Backported to `oparin` in commit 7d40ff4a9e17365932042d85f22430c94f8cd278.

```text
commit 7d40ff4a9e17365932042d85f22430c94f8cd278
Author: Jason Frey <fryguy9@gmail.com>
Date:   Mon Oct 3 14:37:56 2022 -0400

    Merge pull request #520 from agrare/dont_create_vms_from_other_regions
    
    Targeted refresh creates resources from other regions based on events
    
    (cherry picked from commit c086ad02972a832e2415b6c51609b29a7931264e)
```

**Pull request merged**